lambda_src/drivers/destination_s3.py

```diff
@@ -82,15 +83,15 @@ def write_to_s3(bucket: Text, filename: Text, content: AnyStr) -> Dict[Text, Any
     )
 
 
-def initialize(destination, batch_id: Text):
+def initialize(destination: Text, batch_id: Text):
     bucket, prefix = parse_destination_uri(destination)
     content = ''  # We use empty body for creating a folder
-    if not prefix:
-        prefixed_folder_path = f'{DATA_FOLDER_NAME}/{batch_id}/'
-    else:
-        prefixed_folder_path = f'{prefix}/{batch_id}/'
+    # Regex captures characters after and including the rightmost '/' in a path,
+    # which are then replaced with a '/', e.g. '/a/b/c' -> '/a/b/'
+    prefix_folder = re.sub(r'/[^/]*$', "/", prefix) if '/' in prefix else ''
```

Comment:
could we remove the `if` piece since regex replace on an empty string will work regardless?